api/client/application/client_test.go

```diff
@@ -599,107 +657,105 @@ func (s *applicationSuite) TestDestroyRelationId(c *gc.C) {
 		c.Assert(err, jc.ErrorIsNil)
-		c.Assert(called, jc.IsTrue)
 	}
 }
 
 func (s *applicationSuite) TestSetRelationSuspended(c *gc.C) {
-	called := false
-	client := newClient(func(objType string, version int, id, request string, a, result interface{}) error {
-		c.Assert(request, gc.Equals, "SetRelationsSuspended")
-		c.Assert(a, jc.DeepEquals, params.RelationSuspendedArgs{
-			Args: []params.RelationSuspendedArg{
-				{
-					RelationId: 123,
-					Suspended:  true,
-					Message:    "message",
-				}, {
-					RelationId: 456,
-					Suspended:  true,
-					Message:    "message",
-				}},
-		})
-		c.Assert(result, gc.FitsTypeOf, &params.ErrorResults{})
-		*result.(*params.ErrorResults) = params.ErrorResults{
-			Results: []params.ErrorResult{{}, {}},
-		}
-		called = true
-		return nil
-	})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.RelationSuspendedArgs{
+		Args: []params.RelationSuspendedArg{
+			{
+				RelationId: 123,
+				Suspended:  true,
+				Message:    "message",
+			}, {
+				RelationId: 456,
+				Suspended:  true,
+				Message:    "message",
+			}},
+	}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{
+		Results: []params.ErrorResult{{}, {}},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("SetRelationsSuspended", args, result).SetArg(2, results).Return(nil)
+
+	client := application.NewClientFromCaller(mockFacadeCaller)
 	err := client.SetRelationSuspended([]int{123, 456}, true, "message")
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(called, jc.IsTrue)
 }
 
 func (s *applicationSuite) TestSetRelationSuspendedArity(c *gc.C) {
-	called := false
-	client := newClient(func(objType string, version int, id, request string, a, result interface{}) error {
-		c.Assert(request, gc.Equals, "SetRelationsSuspended")
-		c.Assert(a, jc.DeepEquals, params.RelationSuspendedArgs{
-			Args: []params.RelationSuspendedArg{
-				{
-					RelationId: 123,
-					Suspended:  true,
-					Message:    "message",
-				}, {
-					RelationId: 456,
-					Suspended:  true,
-					Message:    "message",
-				}},
-		})
-		c.Assert(result, gc.FitsTypeOf, &params.ErrorResults{})
-		*result.(*params.ErrorResults) = params.ErrorResults{
-			Results: []params.ErrorResult{{}},
-		}
-		called = true
-		return nil
-	})
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	args := params.RelationSuspendedArgs{
+		Args: []params.RelationSuspendedArg{
+			{
+				RelationId: 123,
+				Suspended:  true,
+				Message:    "message",
+			}, {
+				RelationId: 456,
+				Suspended:  true,
+				Message:    "message",
+			}},
+	}
+	result := new(params.ErrorResults)
+	results := params.ErrorResults{
+		Results: []params.ErrorResult{{}, {}},
+	}
+	mockFacadeCaller := mocks.NewMockFacadeCaller(ctrl)
+	mockFacadeCaller.EXPECT().FacadeCall("SetRelationsSuspended", args, result).SetArg(2, results).Return(errors.New("expected 2 results, got 1"))
```

Comment:
The "expected 2 results, got 1" does result from the mock result, it should come from api code when the number of request does not equal the number of results received.